subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -538,3 +506,306 @@ int nrf_cloud_decode_data_endpoint(const struct nrf_cloud_data *input,
 
 	return err;
 }
+
+static int nrf_cloud_encode_service_info_fota(const struct nrf_cloud_svc_info_fota *const fota,
+	cJSON *const svc_inf_obj)
+{
+	if (!svc_inf_obj) {
+		return -EINVAL;
+	}
+
+	if (fota == NULL || !IS_ENABLED(CONFIG_NRF_CLOUD_FOTA)) {
+		if (fota && (fota->application || fota->modem || fota->bootloader)) {
+			LOG_WRN("CONFIG_NRF_CLOUD_FOTA not enabled, setting FOTA array to 'null'");
+		}
+
+		if (json_add_null_cs(svc_inf_obj, JSON_KEY_SRVC_INFO_FOTA) != 0) {
+			return -ENOMEM;
+		}
+	} else if (fota) {
+		int item_cnt = 0;
+		cJSON *array = cJSON_AddArrayToObjectCS(svc_inf_obj, JSON_KEY_SRVC_INFO_FOTA);
+
+		if (!array) {
+			return -ENOMEM;
+		}
+		if (fota->bootloader) {
+			cJSON_AddItemToArray(array, cJSON_CreateString(FOTA_VAL_BOOT));
+			++item_cnt;
+		}
+		if (fota->modem) {
+			cJSON_AddItemToArray(array, cJSON_CreateString(FOTA_VAL_MODEM));
+			++item_cnt;
+		}
+		if (fota->application) {
+			cJSON_AddItemToArray(array, cJSON_CreateString(FOTA_VAL_APP));
+			++item_cnt;
+		}
+
+		if (cJSON_GetArraySize(array) != item_cnt) {
```

Comment:
`item_cnt` is used to ensure each call to `cJSON_AddItemToArray` was successful (instead of checking the return value each time as you mentioned above).
i'm preferring to have one single check for failure.  I can change this if you feel strongly about it.

Reply:
Makes sense, no strong feelings here.